src/main/java/com/teragrep/pth_06/task/hdfs/HdfsRecordConversionImpl.java

```diff
@@ -0,0 +1,125 @@
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.task.hdfs;
+
+import com.teragrep.pth_06.HdfsTopicPartitionOffsetMetadata;
+import com.teragrep.pth_06.avro.SyslogRecord;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.expressions.codegen.UnsafeRowWriter;
+import org.apache.spark.unsafe.types.UTF8String;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+// This class will read the records from avro-files fetched from HDFS with the help of AvroReadImpl and convert them to InternalRow used by pth_06.
+public final class HdfsRecordConversionImpl implements HdfsRecordConversion {
+
+    final Logger LOGGER = LoggerFactory.getLogger(HdfsRecordConversionImpl.class);
+
+    private final boolean stub;
+    private final AvroReadImpl avroReadImpl;
+    private final UnsafeRowWriter rowWriter;
+
+    // Stub object
+    public HdfsRecordConversionImpl(FileSystem fs) {
+        this(fs, new HdfsTopicPartitionOffsetMetadata(new TopicPartition("", 0), 0, "", 0), true);
+    }
+
+    public HdfsRecordConversionImpl(FileSystem fs, HdfsTopicPartitionOffsetMetadata hdfsTopicPartitionOffsetMetadata) {
+        this(fs, hdfsTopicPartitionOffsetMetadata, false);
+    }
+
+    public HdfsRecordConversionImpl(
+            FileSystem fs,
+            HdfsTopicPartitionOffsetMetadata hdfsTopicPartitionOffsetMetadata,
+            Boolean stub
+    ) {
+        this.avroReadImpl = new AvroReadImpl(fs, hdfsTopicPartitionOffsetMetadata);
+        this.stub = stub;
```

Comment:
I think there should be a primary constructor that takes in all the field variables as parameters and make this one a secondary constructor calling the primary one.

Reply:
Solved in commit 8de6ca3a4f90e83f74c91541b6f55be6456a0580